ShowMyData/Pages/Index.cshtml.cs

```diff
@@ -21,11 +21,11 @@ public void OnGet()
         public bool SearchCompleted { get; set; }
         public ICollection<PoliceCrime> policeCrimes { get; set; }
         public ICollection<DrugCrime> drugCrimes { get; set; }
-        public void OnPost()
+        public JsonResult OnPost()
```

Comment:
This is changing functionality.  As the group has it now, an HTML page is returned.  You're changing this to return JSON.  The group may prefer what they had originally.  For a post method, I would not anticipate a JSON response.